fem/coefficient.hpp

```diff
@@ -624,47 +766,66 @@ class MatrixFunctionCoefficient : public MatrixCoefficient
       mat.SetSize(0);
    }
 
+   /// Evaluate the matrix coefficient at @a ip.
    virtual void Eval(DenseMatrix &K, ElementTransformation &T,
                      const IntegrationPoint &ip);
 
    virtual ~MatrixFunctionCoefficient() { }
 };
 
+
+
+/** @brief Matrix coefficient defined by an matrix of scalar coefficients.
+    Coefficients that are not set will evaluate to zero in the vector.  The
+    of coefficient is stored as a flat Array with indexing (i,j) -> i*width+j.
+    This object takes ownership of the array of coefficients inside it and
```

Comment:
This is not true.  The ownership can be set independently for each matrix entry.  The array is owned by the `MatrixArrayCoefficient` but the scalar `Coefficients` within it may not be owned by this object.

Reply:
It looks like it's optional now, but I think it was the case when I originally wrote this.  Great catch, I'll fix it.